lib/libc/minimal/include/stdlib.h

```diff
@@ -30,8 +30,12 @@ void *bsearch(const void *key, const void *array,
 	      size_t count, size_t size,
 	      int (*cmp)(const void *key, const void *element));
 
-void qsort_r(void *base, size_t nmemb, size_t size,
-	     int (*compar)(const void *, const void *, void *), void *arg);
+typedef int (*compar_fn_t)(const void *, const void *);
```

Comment:
Generally, we do not want to add new typedefs into headers.

Reply:
And why is that? It makes type checking easy, and the match between prototypes and usages can be checked at compile time.

Reply:
To avoid collisions in the global namespace especially in the case of standard library headers

Reply:
I see that, but as you state, this is a standard library, and standard libraries do provide typedefs in their header files. Those two typedefs are actually the same as from glibc stdlib.h, except for removal of __.
What is needed in order to keep these typedefs? Will that be adding the _'s?
Or shall the typedefs be removed completely?

Reply:
I'm ok with this changes to stdlib.h as long as the typedefs can be removed.

Reply:
I have removed the typedefs.